### ISSUE TITLE
Serializing sources and transformations 

### ISSUE THREAD

**Issue by bogovicj** [saalfeldlab/bigwarp#158]
Part of this issue came from: https://github.com/saalfeldlab/bigwarp/issues/148#issuecomment-1235866060

## Serializing sources

We'll aim to store a url that will enable bigwarp serialize a source and re-open it on startup.
Inspired by neuroglancer, we thought a non-standard url of the form:
`typeofstore:pathToFileOrUrl?dataset&other options`

where `pathToFileOrUrl` may contain a protocol, for example:

* `n5:file:///path/to/my.n5?volume`
* `zarr:s3:///path/to/my.zarr?images/a/s0`
* `h5:path/to/data.hdf5?data`

We omit the initial protocol for "regular" files, for example:
* `file:///home/a/b.tif`
* `/tmp/data.nrrd`

Questions - 
* Do we require an "initial" n5/ zarr / h5 protocol? or is it suggested but not required.
  * i.e. is this allowed `file:///path/to/foo.h5?img`
  * The protocol will matter most (I think) in enabling opening datasets from resources that are valid zarr/n5/h5 stores but don't have the usual extensions.
 
## Referencing serialized transformations

We will use the [ome-ngff transformation spec](https://github.com/bogovicj/ngff/tree/coord-transforms), but will need to be able to reference the store that contains them, and where they can be found in the store.

We can reference stores and groups / datasets in the store using the same method as serializing sources

For example if a filesystem has the file /path/to/my.n5/volume/attributes.json`, and it contains:

```json
{
    "coordinateTransformations" : [
        { "type" : "scale","scale" : [1,1] },
        { "type" : "scale", "scale" : [2,2] }
    ],
    "scale" : { "type" : "scale", "scale" : [3,3] },
    "a" : {
       "b" : { "type" : "scale", "scale" : [4,4] }
    }
}

```

Then:
 * `n5:file:///path/to/my.n5?volume&coordinateTransformations[0]` refers to `scale[1,1]`
 * `n5:file:///path/to/my.n5?volume&coordinateTransformations[1]` refers to `scale[2,2]`
 * `n5:file:///path/to/my.n5?volume&scale` refers to `scale[3,3] `
 *  `n5:file:///path/to/my.n5?volume&a.b` refers to `scale[4,4]`

We could have the default attribute be `coordinateTransformations` meaning that: 

 * `n5:file:///path/to/my.n5?volume&[0]` refers to `scale[1,1]`
 * `n5:file:///path/to/my.n5?volume&[1]` refers to `scale[2,2]`

**Comment by bogovicj** [comment]
The above scheme is out of date.  Up to date scheme here:
https://github.com/ome/ngff/issues/144#issuecomment-1314299599

we'll track progress supporting the scheme above here.

**Comment by bogovicj** [comment]
I found an issue on loading sources from a settings file in the current url branch, at this commit, for example:
https://github.com/saalfeldlab/bigwarp/commit/d730e9f42599ffeb05823312a4ce06d167518790

Summary: loading 3d sources into an empty bigwarp using the 2d transform `TransformEventHandler`, and unfortunately, [it is final and private](https://github.com/bigdataviewer/bigdataviewer-core/blob/master/src/main/java/bdv/viewer/ViewerPanel.java#L137).  We probably need to create bigwarp viewer frames in this case.

@cmhulbert , can you have a look?

**Comment by cmhulbert** [comment]
@bogovicj I tried to duplicate with no success. I added a [main method to SerializationTest](https://github.com/saalfeldlab/bigwarp/blob/url/src/test/java/bigwarp/SerializationTest.java#L492) which allows you to test this manually. Let me know if it faithfully duplicates the steps you listed above, and if you see the same bahavior I do (i.e. 3d-navigation properly working)

**Comment by bogovicj** [comment]
Implemented as of bigwarp_fiji-9.0.0 de10b7a61bab9c39b28d59279814c690573db140
